cli/src/commands/invite/claim.rs

```diff
@@ -69,9 +78,58 @@ pub async fn main(args: Args) -> anyhow::Result<()> {
             let ctx = step4_device(ctx).await?;
             save_device(ctx, save_mode).await
         }
-        AnyClaimRetrievedInfoCtx::ShamirRecovery(_) => Err(anyhow::anyhow!(
-            "Shamir recovery invitation is not supported yet"
-        )),
+        AnyClaimRetrievedInfoCtx::ShamirRecovery(ctx) => {
+            let mut pick_ctx = ctx;
+
+            let mut device_ctx = loop {
+                let ctx = shamir_pick_recipient(&pick_ctx)?;
+                let ctx = step1_shamir(ctx).await?;
+                let ctx = step2_shamir(ctx).await?;
+                let ctx = step3_shamir(ctx).await?;
+                let share_ctx = step4_shamir(ctx).await?;
+                let maybe = pick_ctx.add_share(share_ctx)?;
```

Comment:
We don't allow the user to add more share?

Reply:
The number of share is setup during the creation of the device with the command `shared-recovery create`